LanguageFeatures/Patterns/pattern_uses_A02_t02.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// The static type of a switch expression is the least upper bound of the
+/// static types of all of the case expressions.
+///
+/// @description Check that it is a compile-time error to assign a switch
+/// expression to a type which is not the least upper bound of the static types
```

Comment:
We should probably say something like 'to a type `T` such that the standard upper bound of the static types of all its case expressions is not assignable to `T`'.